### ISSUE TITLE
How to hotswap GUI components in JavaFX

### ISSUE THREAD

**Issue by kllp** [HotswapProjects/HotswapAgent#290]
Do you know a trick to make JavaFX graphical components reload when they are changed?
Is it possible?

I managed to make HotswapAgent work for commandline programs but not for the GUI of simple JavaFX apps.
Regarding graphical hotswapping I only found examples for web applications or TornadoFX.

**Comment by cristian-spiescu** [comment]
I worked a bit with JavaFX and HotswapAgent too. I think that what you are saying is that: 1) your app is running; 2) you modify some code; 3) you expect something to be different at the UI; e.g. a modified DOM tree of the current screen. Am I right?

What I did in my case: I have a function that creates the DOM (i.e. the components of the current screen). And in the UI I have a button that calls this function. Basically it recreates the screen. And according to the flow above, I introduces a 2.5) step, where I actually press on that button.
